Zend/zend_compile.c

```diff
@@ -3025,7 +3067,11 @@ void zend_compile_assign(znode *result, zend_ast *ast) /* {{{ */
 				/* list($a, $b) = $a should evaluate the right $a first */
 				zend_compile_simple_var_no_cv(&expr_node, expr_ast, BP_VAR_R, 0);
 			} else {
-				zend_compile_expr(&expr_node, expr_ast);
+				if (zend_compile_list_assign_requires_w(var_ast) && zend_is_variable(expr_ast)) {
+					zend_compile_var(&expr_node, expr_ast, BP_VAR_W);
+				} else {
+					zend_compile_expr(&expr_node, expr_ast);
```

Comment:
I'm wondering if it wouldn't be better to throw a compile time error in this case. After all things like `$foo =& 42` are parse errors, so I would expect `list(&$foo) = [42]` to be a compile error.

Reply:
I can change.  I guess they are kind of synonymous, attempt to assign to a non referenceable value.